### PR TITLE
[DEV-3099] Award History Table, Transaction history Tab

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
Implements the Award History Table on _all_ AS2.0 pages for the Transaction history pages. 

**Technical details:**
- Integrates w/ API that renders the count value for transactions/subawards/fed_accounts for given award
- Updates Contracts and Financial Assistance Content components
- Updates TableSection Component w/ API call
- Reuses styles for count display from IDV Child Awards table

**JIRA Ticket:**
[DEV-3099](https://federal-spending-transparency.atlassian.net/browse/DEV-3099)

**Mockup:**
https://bahdigital.invisionapp.com/share/U9IAAWH7MA6#/296011639_Award_Summary_2-0_-_Contract

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] All `componentWillReceiveProps`, `componentWillMount`, and `componentWillUpdate` in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3341](https://federal-spending-transparency.atlassian.net/browse/DEV-3341)
- [x] Design review (if applicable)
- [x] #1257 merged

### CODE REVIEW

**Comment by MaxwellKendall** [comment]
@ebdabbs & @jonhill13 

Most of these changes are updates to the import paths for quite a few files that I moved into a shared directory.

The real changes relevant to the ticket are in these files:
- `TransactionsTable`
- `ContractContent`
- `FinanacialAssistanceContent`
- `AwardHistory`
- `dataMapping/transactionHistoryTable`

**Comment by AGuyNamedMarco** [comment]
@MaxwellKendall I mentioned this in the thread for #1268 but thought it was relevant here as well.

> We can remove the "Sub-Awards" tab from the Loan template, since as it turns out, Loans never have sub-loans (if they do, it's an error in the data).

File: src/js/components/awardv2/contract/ContractContent.jsx [hunk 1]
```diff
@@ -16,56 +16,66 @@ import AwardPageWrapper from '../shared/AwardPageWrapper';
...
-    render() {
-        const { overview, awardId, jumpToSection } = this.props;
-        const glossarySlug = glossaryLinks[overview.type];
-        const glossaryLink = glossarySlug
-            ? `/#/award_v2/${awardId}?glossary=${glossarySlug}`
-            : null;
-        const awardAmountData = Object.create(BaseAwardAmounts);
-        awardAmountData.populate(overview, overview.category);
+const ContractContent = ({ awardId, overview, jumpToSection }) => {
+    const [activeTab, setActiveTab] = useState("transaction");
```
Comment (ebdabbs) [inline src/js/components/awardv2/contract/ContractContent.jsx]: Nice!

File: src/js/components/awardv2/contract/ContractContent.jsx [hunk 2]
```diff
@@ -16,56 +16,66 @@ import AwardPageWrapper from '../shared/AwardPageWrapper';
...
-                        awardType={this.props.overview.category}
-                        jumpToSection={this.props.jumpToSection}
-                        awardOverview={awardAmountData} />
-                    <ComingSoonSection title="Description" includeHeader />
+    const awardAmountData = Object.create(BaseAwardAmounts);
+    awardAmountData.populate(overview, overview.category);
+
+    const jumpToTransactionHistoryTable = () => {
+        setActiveTab('transaction');
+        jumpToSection("award-history");
```
Comment (ebdabbs) [inline src/js/components/awardv2/contract/ContractContent.jsx]: Recommend single quotes throughout for consistency

File: src/js/components/awardv2/idv/amounts/AggregatedAwardAmounts.jsx [hunk 3]
```diff
@@ -67,16 +72,7 @@ export default class AggregatedAwardAmounts extends React.Component {
...
-                    onClick={this.jumpToReferencedAwardsTable}
-                    className="award-viz__button">
-                    <div className="award-viz__link-icon">
-                        <Table />
-                    </div>
-                    <div className="award-viz__link-text">
-                        View award orders table
-                    </div>
-                </button>
+                <JumpToSectionButton linkText="View award orders table" onClick={this.jumpToReferencedAwardsTable} icon="table" />
```
Comment (ebdabbs) [inline src/js/components/awardv2/idv/amounts/AggregatedAwardAmounts.jsx]: 👍

File: src/js/components/awardv2/shared/awardHistorySection/AwardHistory.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/**
+  * AwardInfo.jsx
```
Comment (ebdabbs) [inline src/js/components/awardv2/shared/awardHistorySection/AwardHistory.jsx]: changed to AwardHistory.jsx

File: src/js/helpers/awardHistoryHelper.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/**
+  * idvHelper.js
```
Comment (ebdabbs) [inline src/js/helpers/awardHistoryHelper.js]: awardHistoryHelper.js

**Comment by MaxwellKendall** [comment]
> @MaxwellKendall I mentioned this in the thread for #1268 but thought it was relevant here as well.
> 
> > We can remove the "Sub-Awards" tab from the Loan template, since as it turns out, Loans never have sub-loans (if they do, it's an error in the data).

@AGuyNamedMarco -- addressed this in #1268 w/ [this commit](https://github.com/fedspendingtransparency/usaspending-website/pull/1268/commits/b5fd36b95353e2ede5c0a9a7ccab1bd75fea9986)

**Comment by MaxwellKendall** [comment]
Going to merge #1268 which contains this PR

**Pull request closed without merging**